app/src/main/java/tttTDD/Java/Board.java

```diff
@@ -1,44 +1,90 @@
 package tttTDD.Java;
 
-import org.checkerframework.common.value.qual.IntRange;
-
-import java.lang.reflect.Array;
+import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 public class Board {
+    private final int DEFAULT_SIZE = 3;
     private final String[] board;
+    private int numberOfSpaces;
+    private int[][] winningSets = {
+            {0, 1, 2},
+            {3, 4, 5},
+            {6, 7, 8},
+            {0, 3, 6},
+            {1, 4, 7},
+            {2, 5, 8},
+            {0, 4, 8},
+            {2, 4, 6},
+    };
 
     public Board() {
-        this.board = new String[] {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
+        this.board = initializeBoard(DEFAULT_SIZE);
     }
 
     public Board(int boardSize) {
-        this.board = new String[boardSize * boardSize];
-        for (int i = 0; i < this.board.length; i++) {
-            this.board[i] = String.valueOf(i + 1);
-        }
+        this.board = initializeBoard(boardSize);
     }
 
-    public String[] getBoard() {
-        return this.board;
+    public String getSpace(int location) {
+        return isLocationValid(location) ? null : this.board[location];
     }
 
-    public String getSpace(int location) {
-        return this.board[location];
+    public int getFirstAvailableMove() {
+        String[] availableMoves = getAvailableMoves(board);
+        return convertToLocation(availableMoves[0]);
     }
 
-    public void updateBoard(int location, String marker) throws Exception {
+    public boolean updateBoard(int location, String marker) {
         if (isSpaceAvailable(location)) {
             this.board[location] = marker;
-        } else {
-            throw new Exception("The space is already occupied");
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean hasWinningSet() {
+        for (int[] set : winningSets) {
+            if (board[set[0]].equals(board[set[1]]) && board[set[1]].equals(board[set[2]])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String[] initializeBoard(int boardSize) {
+        numberOfSpaces = boardSize * boardSize;
+        String[] board = new String[numberOfSpaces];
+        for (int i = 0; i < board.length; i++) {
+            board[i] = String.valueOf(i + 1);
```

Comment:
What benefits do we derive from having a different value in each uninhabited square?

Reply:
The benefits derived is to distinguish each space from one another once a board is rendered if there's only a keyboard available to enter the appropriate cell.

Reply:
> distinguish each space from one another

Gotcha, the stringified counting numbers indicate the universe of allowed inputs that you envision eventually presenting in a text-based user interface.

If that's the case (that we're optimizing the storage of this to match its display), though, why does the updateBoard method take a position as a zero-based integer?

In other words, why are we doing translation of the more efficient and non-specific representation of a tic-tac-toe square on the inputs to this class, but not on the outputs?